packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -110,7 +110,7 @@ class SocketReference {
             // Treat all errors as recoverable, and rely on joinSession / reconnection flow to
             // filter out retryable vs. non-retryable cases.
             const error = errorObjectFromSocketError(socketError);
-            error.canRetry = true;
+            error.canRetry = error.canRetry ?? true;
```

Comment:
I do not think this is correct. What if token expired and we are getting 401/403 here?
This is in no way non-retryable error, we need to go through reconnection process to know for sure if client has lost permissions vs. just need to refresh token.
In my view this is dangerous change in current flow where the invariant we have today - all errors can be learned on connection path. Same should apply to epoch failures. If we can't learn about epoch changes there, then we have much bigger problem, as we might be loading from very old snapshot and need quickly realize epoch bump through joinSession/connect.

Reply:
Removed the change.